x/mongo/driver/topology/pool.go

```diff
@@ -583,6 +588,13 @@ func (p *pool) checkIn(conn *connection) error {
 		})
 	}
 
+	// Decrement the number of in-use connections once we know we have a non-nil connection that
+	// came from this pool. Do this in checkIn() instead of checkInNoEvent() because the latter is
+	// called by createConnections() and is not necessarily a check-in of an in-use connection. Use
+	// an int64 instead of a uint64 to mitigate the impact of any possible bugs that could cause the
+	// uint64 to underflow, which would effectively make the server unselectable.
```

Comment:
I agree with the rationale for using an int64 rather than an uint64. I think that is a good precaution.

One suggested edit:

```suggestion
	// uint64 to underflow, which would make the server much less selectable.
```

A server could still be selected if it has the lowest average latency.

Reply:
Good suggestion, will update.

Reply:
Updated.